src/governance-classes.cpp

```diff
@@ -512,9 +509,31 @@ CSuperblock(uint256& nHash)
     DBG( cout << "CSuperblock Constructor End" << endl; );
 }
 
-void 
-CSuperblock::
-ParsePaymentSchedule(std::string& strPaymentAddresses, std::string& strPaymentAmounts)
+bool CSuperblock::IsValidBlockHeight(int nBlockHeight)
+{
+    // SUPERBLOCKS CAN HAPPEN ONLY after hardfork and only ONCE PER CYCLE
+    return nBlockHeight >= Params().GetConsensus().nSuperblockStartBlock &&
+            ((nBlockHeight % Params().GetConsensus().nSuperblockCycle) == 0);
+}
+
+CAmount CSuperblock::GetPaymentsLimit(int nBlockHeight)
+{
+    const Consensus::Params& consensusParams = Params().GetConsensus();
+
+    if(!IsValidBlockHeight(nBlockHeight))
+        return 0;
+
+    // min subsidy for high diff networks and vice versa
+    int nBits = consensusParams.fPowAllowMinDifficultyBlocks ? UintToArith256(consensusParams.powLimit).GetCompact() : 1;
+    CAmount nBlockSubsidy = GetBlockSubsidy(nBits, nBlockHeight, consensusParams);
+    // 10% of all blocks issued during the cycle goes to superblock
+    CAmount nPaymentsLimit = nBlockSubsidy * consensusParams.nSuperblockCycle / 10;
```

Comment:
I don't think this calculation is quite exact because GetBlockSubsidy already returns a value reduced by 10% to account for the budget/superblock payments.  So I think it should be:

(conceptually, maybe better not to mix doubles and int64's in real code):
nPaymentsLimit = 1.1 * nBlockSubsidy * consensusParams.nSuperblockCycle / 10;

or

nPaymentsLimit = nBlockSubsidy * consensusParams.nSuperblockCycle / 10 + nBlockSubsidy * consensusParams.nSuperblockCycle / 100